test/e2e/v3/plugin_cluster_test.go

```diff
@@ -0,0 +1,276 @@
+			true,
+			"secrets", "webhook-server-cert")
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("validating that the Prometheus manager has provisioned the Service")
+	EventuallyWithOffset(1, func() error {
+		_, err := kbc.Kubectl.Get(
+			false,
+			"Service", "prometheus-operator")
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("validating that the ServiceMonitor for Prometheus is applied in the namespace")
+	_, err = kbc.Kubectl.Get(
+		true,
+		"ServiceMonitor")
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	By("validating that the mutating|validating webhooks have the CA injected")
+	verifyCAInjection := func() error {
+		mwhOutput, err := kbc.Kubectl.Get(
+			false,
+			"mutatingwebhookconfigurations.admissionregistration.k8s.io",
+			fmt.Sprintf("e2e-%s-mutating-webhook-configuration", kbc.TestSuffix),
+			"-o", "go-template={{ range .webhooks }}{{ .clientConfig.caBundle }}{{ end }}")
+		ExpectWithOffset(2, err).NotTo(HaveOccurred())
+		// check that ca should be long enough, because there may be a place holder "\n"
+		ExpectWithOffset(2, len(mwhOutput)).To(BeNumerically(">", 10))
+
+		vwhOutput, err := kbc.Kubectl.Get(
+			false,
+			"validatingwebhookconfigurations.admissionregistration.k8s.io",
+			fmt.Sprintf("e2e-%s-validating-webhook-configuration", kbc.TestSuffix),
+			"-o", "go-template={{ range .webhooks }}{{ .clientConfig.caBundle }}{{ end }}")
+		ExpectWithOffset(2, err).NotTo(HaveOccurred())
+		// check that ca should be long enough, because there may be a place holder "\n"
+		ExpectWithOffset(2, len(vwhOutput)).To(BeNumerically(">", 10))
+
+		return nil
+	}
+	EventuallyWithOffset(1, verifyCAInjection, time.Minute, time.Second).Should(Succeed())
+
+	By("creating an instance of the CR")
+	// currently controller-runtime doesn't provide a readiness probe, we retry a few times
+	// we can change it to probe the readiness endpoint after CR supports it.
+	sampleFile := filepath.Join("config", "samples",
+		fmt.Sprintf("%s_%s_%s.yaml", kbc.Group, kbc.Version, strings.ToLower(kbc.Kind)))
+	EventuallyWithOffset(1, func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", sampleFile)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("applying the CRD Editor Role")
+	crdEditorRole := filepath.Join("config", "rbac",
+		fmt.Sprintf("%s_editor_role.yaml", strings.ToLower(kbc.Kind)))
+	EventuallyWithOffset(1, func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", crdEditorRole)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("applying the CRD Viewer Role")
+	crdViewerRole := filepath.Join("config", "rbac", fmt.Sprintf("%s_viewer_role.yaml", strings.ToLower(kbc.Kind)))
+	EventuallyWithOffset(1, func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", crdViewerRole)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("validating that the created resource object gets reconciled in the controller")
+	metricsOutput := curlMetrics(kbc)
```

Comment:
Metrics are now being used to confirm a successful reconcile since the "Successfully Reconcile" log message was [removed](https://github.com/kubernetes-sigs/controller-runtime/pull/1096).